### PR TITLE
merge set_satellite_options and set_custom_provision_type into set_custom_provision_task_options.rb

### PR DESCRIPTION
**Pull request by itewk**

### CODE REVIEW

File: Automate/RedHatConsulting_Satellite6/Service/Provisioning/StateMachines/Methods.class/__methods__/set_custom_provision_task_options.rb [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    
+      # query satellite for the domain id
+      satellite_domains = satellite_api.resource(:domains).call(:index)['results']
+      satellite_domain  = satellite_domains.find { |domain| domain['name'] == domain_name }
+      error("Could not find Satellite Domain with name: '#{domain_name}'") if satellite_domain.nil?
+    
+      # set the additional dialog option
+      custom_additional_values[:satellite_domain_id] = satellite_domain['id']
+    else
+      error("One of <satellite_domain_id, domain_name> must be suplied as dialog options.")
```
Comment (henrygrover) [inline Automate/RedHatConsulting_Satellite6/Service/Provisioning/StateMachines/Methods.class/__methods__/set_custom_provision_task_options.rb]: suplied -> supplied

File: Automate/RedHatConsulting_Satellite6/Service/Provisioning/StateMachines/Methods.class/__methods__/set_custom_provision_task_options.rb [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+#
+@DEBUG = false
+
+require 'apipie-bindings'
+
+# IMPLIMENTORS: intended to be overwritten
+#
+# @param dialog_options Hash Dialog options set by user when creating the service.
+#
+# @return Hash of custom vm_fileds to set on any vm provisioning requests created for the service being created
```
[the remark this answers is not in the document]
Comment (itewk) [reply Automate/RedHatConsulting_Satellite6/Service/Provisioning/StateMachines/Methods.class/__methods__/set_custom_provision_task_options.rb]: fixed.

File: Automate/RedHatConsulting_Satellite6/Service/Provisioning/StateMachines/Methods.class/__methods__/set_custom_provision_task_options.rb (hunk 1 above)
Comment (itewk) [reply Automate/RedHatConsulting_Satellite6/Service/Provisioning/StateMachines/Methods.class/__methods__/set_custom_provision_task_options.rb]: fixed.

**Comment by itewk** [comment]
@henrygrover fixed your issues, re-approve?

**Pull request merged**